Intersect.Client/Entities/Player.cs

```diff
@@ -2560,27 +2678,77 @@ private partial class TargetInfo
             {
-                if (!Controls.KeyDown(Control.TurnAround) || direction != Globals.Me.MoveDir)
+                if (!Controls.KeyDown(Control.TurnAround) || direction != (int)Globals.Me.MoveDir)
                 {
                     continue;
                 }
 
                 // Turn around and hold the player in place if the requested direction is different from the current one.
                 if (!Globals.Me.IsMoving && Dir != Globals.Me.MoveDir)
                 {
-                    Dir = (byte)Globals.Me.MoveDir;
-                    PacketSender.SendDirection(Dir);
-                    Globals.Me.MoveDir = -1;
+                    Dir = Globals.Me.MoveDir;
+                    PacketSender.SendDirection((byte)Dir);
+                    Globals.Me.MoveDir = Direction.None;
                 }
 
                 // Hold the player in place if the requested direction is the same as the current one.
                 if (!Globals.Me.IsMoving && Dir == Globals.Me.MoveDir)
                 {
-                    Globals.Me.MoveDir = -1;
+                    Globals.Me.MoveDir = Direction.None;
+                }
+            }
+        }
+        
+        // Checks if the target is at the opposite direction of the current player's direction.
+        // The comparison also takes into account whether diagonal movement is enabled or not.
+        private static bool IsTargetAtOppositeDirection(Direction currentDir, Direction targetDir)
+        {
+            if (Options.Instance.MapOpts.EnableDiagonalMovement)
+            {
+                // If diagonal movement is enabled, check for the opposite direction for 8 directions.
+                switch (currentDir)
+                {
+                    case Direction.Up:
+                        return targetDir == Direction.Down || targetDir == Direction.DownLeft ||
+                               targetDir == Direction.DownRight;
+                    case Direction.Down:
+                        return targetDir == Direction.Up || targetDir == Direction.UpLeft ||
+                               targetDir == Direction.UpRight;
+                    case Direction.Left:
+                        return targetDir == Direction.Right || targetDir == Direction.UpRight ||
+                               targetDir == Direction.DownRight;
+                    case Direction.Right:
+                        return targetDir == Direction.Left || targetDir == Direction.UpLeft ||
+                               targetDir == Direction.DownLeft;
+                    case Direction.UpLeft:
+                        return targetDir == Direction.DownRight;
+                    case Direction.UpRight:
+                        return targetDir == Direction.DownLeft;
+                    case Direction.DownLeft:
+                        return targetDir == Direction.UpRight;
+                    case Direction.DownRight:
+                        return targetDir == Direction.UpLeft;
```

Comment:
I feel like if Left/Right/Up/Down can "see" the diagonal, the diagonals should be able to "see" the cardinal directions, these switch cases are not consistent

Reply:
self reminder: double check this tomorrow.